common/utils/src/main/resources/error/error-conditions.json

```diff
@@ -1914,6 +1914,29 @@
     ],
     "sqlState" : "22012"
   },
+  "INTERVAL_ERROR" : {
+    "message" : [
+      "Interval error."
+    ],
+    "subClass" : {
+      "ILLEGAL_DAY_OF_WEEK" : {
+        "message" : [
+          "Illegal input for day of week: <string>."
+        ]
+      },
+      "INTERVAL_PARSING" : {
+        "message" : [
+          "Error parsing interval <interval> string: <msg>."
```

Comment:
Let's don't include arbitrary message from `cause`. Just assign it to the `cause` field.

Reply:
Can we move this class and make it a subclass of the `INVALID_INTERVAL_FORMAT`?

Reply:
Is then the error message sufficient for this error class? If we remove the `msg` parameter, the error message will be looking something like this: `Error parsing interval year-month.`.

Reply:
There is an example in the `DDLParserSuite`, test is called `Invalid interval term should throw AnalysisException`